src/Logging/Logging.Batching/BatchLogService.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace CodeMonkeys.Logging.Batching
+{
+    public abstract class BatchLogService<TOptions> : ScopedLogService<TOptions>
+        where TOptions : BatchLogOptions, new()
+    {
+        private BlockingCollection<LogMessage> _queue;
+
+        public override bool IsEnabled
+        {
+            get => base.IsEnabled;
+            protected set
+            {
+                if (value)
+                    Run();
+
+                base.IsEnabled = value;
+            }
+        }
+
+        protected BatchLogService(string context) 
+            : base(context)
+        {
+            Run();
+        }
+
+        protected abstract Task PublishMessageBatch(IEnumerable<LogMessage> messageBatch);
+
+        protected override void PublishMessage(LogMessage message)
+        {
+            if (_queue.IsAddingCompleted)
+                return;
+
+            try
+            {
+                _queue.Add(message);
+            }
+            catch { }
+        }
+
+        private void Run()
+        {
+            _queue = Options.QueueCapacity == null ?
+                new BlockingCollection<LogMessage>(new ConcurrentQueue<LogMessage>()) :
+                new BlockingCollection<LogMessage>(
+                    new ConcurrentQueue<LogMessage>(),
+                    Options.QueueCapacity.Value);
+
+            Task.Run(ProcessingLoop);
+        }
+
+        private async Task ProcessingLoop()
```

Comment:
I‘d prefer `ProcessLoop()`

Reply:
I prefer ProcessingLoop because it sounds more like "a loop in which processing is done." ProcessLoop sounds more like: "process the loop" in my opinion